.github/workflows/ca-sequential-test.yml

```diff
@@ -1836,6 +1836,551 @@ jobs:
 
           diff expected actual
 
+      ####################################################################################################
+      # Enroll a cert with newLegacy
+      #
+      # This should work like the legacy .
+      - name: Switch to newLegacy
+        run: |
+          # switch cert request ID generator to RSNv3
+          BEGIN_SERIAL=$(docker exec pki pki-server ca-config-show dbs.beginSerialNumber)
+          docker exec pki pki-server ca-config-set dbs.beginSerialNumber 0x$BEGIN_SERIAL
+          END_SERIAL=$(docker exec pki pki-server ca-config-show dbs.endSerialNumber)
+          docker exec pki pki-server ca-config-set dbs.endSerialNumber 0x$END_SERIAL
+          docker exec pki pki-server ca-config-set dbs.request.id.generator newLegacy
+
+          docker exec pki pki-server ca-config-set dbs.cert.id.generator newLegacy
+
+          docker exec pki pki-server ca-range-update
+          # restart CA subsystem
+          docker exec pki pki-server restart --wait
+          
+
+      - name: Check request range config
+        run: |
+          docker exec pki pki-server ca-config-find \
+              | grep \
+                  -e dbs.beginRequestNumber \
+                  -e dbs.endRequestNumber \
+                  -e dbs.requestCloneTransferNumber \
+                  -e dbs.requestIncrement \
+                  -e dbs.requestLowWaterMark \
+              | tee actual
+
+          # request range should be the same
+          cat > expected << EOF
+          dbs.beginRequestNumber=31
+          dbs.endRequestNumber=40
+          dbs.requestCloneTransferNumber=5
+          dbs.requestIncrement=10
+          dbs.requestLowWaterMark=5
+          EOF
+
+          diff expected actual
+
+      - name: Check cert range config
+        run: |
+          docker exec pki pki-server ca-config-find \
+              | grep \
+                  -e dbs.beginSerialNumber \
+                  -e dbs.endSerialNumber \
+                  -e dbs.serialCloneTransferNumber \
+                  -e dbs.serialIncrement \
+                  -e dbs.serialLowWaterMark \
+              | tee actual
+
+          # cert range should be the same
+          cat > expected << EOF
+          dbs.beginSerialNumber=0x27
+          dbs.endSerialNumber=0x36
+          dbs.serialCloneTransferNumber=8
+          dbs.serialIncrement=10
+          dbs.serialLowWaterMark=8
```

Comment:
Should we change these params to hex too (especially the increment)?

Reply:
with this change the increment for requests and certs are aligned. If I should install a new CA, I think I would use same values for both. Here we are migrating from an existing configuration and we could decide to have different increments. Unless do you mean the increment in hex but with the amount of 10 (0xa) in this case.

Reply:
Before the migration `dbs.serialIncrement=10` meant 10 hex or 16 decimal, but after the migration to the new ID generator it becomes 10 decimal, so we're mixing different formats in one configuration (i.e. for cert ID generator) and it will actually change the size of the ranges. If there's a 3rd party application relying on these params or ranges (although it's not recommended) they might have to change their application too. In the future we might automate the migration process, and we'll probably modify these params the same way (i.e. becoming all hex for cert ID generator) so this test should reflect the same process.